### PR TITLE
Blocks propagation

### PR DESCRIPTION
**Pull request by NikVolf**
sends new known hashes to peers that probably have not got em

### CODE REVIEW

**Comment by NikVolf** [comment]
ah when writing tests in other branch discovered that packet for all propagades should contain difficulty as well

File: ethcore/src/block_queue.rs [hunk 1]
```diff
@@ -38,6 +38,8 @@ pub struct BlockQueueInfo {
 	pub verified_queue_size: usize,
 	/// Number of blocks being verified
 	pub verifying_queue_size: usize,
+	/// Indicates queue is empty
+	pub empty: bool
```
Comment (gavofyork) [inline ethcore/src/block_queue.rs]: is this really a necessary optimisation? (why not `fn empty(&self) -> bool { verification.verified.is_empty() && verification.unverified.is_empty() && verification.verifying.is_empty() }`?

Comment (NikVolf) [inline ethcore/src/block_queue.rs]: function for BlockQueueInfo?
seems reasonable
I guess 'full' should be function as well then, right?

Comment (gavofyork) [inline ethcore/src/block_queue.rs]: indeed.

Comment (NikVolf) [inline ethcore/src/block_queue.rs]: - resolved

**Pull request merged**